libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -136,13 +132,14 @@ type segmentHeader struct {
 	frameCount uint32
 }
 
-const currentSegmentVersion = 1
+// segmentHeaderSizeV0 schemaVersion 0 header size, uint32 for version
+const segmentHeaderSizeV0 = 4
 
-// Segment headers are currently a 4-byte version plus a 4-byte frame count.
-// In contexts where the segment may have been created by an earlier version,
-// instead use (queueSegment).headerSize() which accounts for the schema
-// version of the target segment.
-const segmentHeaderSize = 8
+// segmentHeaderSizeV1 schemaVersion 1 header size, uint32 for version, uint32 for count
+const segmentHeaderSizeV1 = 8
+
+// segmentHeaderSizeV2 schemaVersion 2 header size uint32 for version
+const segmentHeaderSizeV2 = 4
```

Comment:
Why does this header size remove the frameCount field? That is part of the queue's bookkeeping and is used in metrics reporting.

It seems to me `segmentHeader` instead needs to grow to also include flags for encryption and compression (since these can in principle vary for every segment if the process is restarted with a different configuration).

Reply:
back to the design doc.  The concern that was brought up was that it was outside the encryption, and as such an attacker could modify it to a known value, making it's use in the book keeping suspect.  Moving it inside encryption has it's own issues, you can't put it at the front because you can't seek back in the encryption stream and re-write the first bits, that throws everything off.  You could add it to the end, but then you have to have decrypted the whole thing anyway, in which case it behaves a lot like schema version 0.